sample_app/templates/dynamic_formset_example.html

```diff
@@ -0,0 +1,197 @@
+
+        const newFormHtml = formTemplateNode.outerHTML
+            .replace(fieldFormRegex, `${formsetPrefix}-${newFormIndex}-`)
+            // note the fragment IDs don't have a '-' suffix
+            .replace(fragmentsFormRegex, `fragment-${formsetPrefix}-${newFormIndex}`);
+
+
+        return newFormHtml
+    }
+
+    createNewFormNode(formIndex) {
+        // override in case, e.g. index needs to be used in text
+
+
+        // It would be easier to write the parent's innerHTML
+        // but that would risk losing event listeners, potentially the addButton! 
+        // Also, since we don't know what nodeType the form is
+        // we use a dummy parent rather than createElement(unknownNodeType)
+        const dummyContainer = document.createElement('div')
+        dummyContainer.innerHTML = this.getNewHtmlForForm(formIndex)
+        const newForm = dummyContainer.children[0]
+
+        return newForm;
+    }
+
+    addForm(e) {
+        e.preventDefault();
+
+        const previousNumForms = this.getFormNodes().length
+        const newFormIndex = previousNumForms; // indexing starts at 0
+        const newNumForms = previousNumForms + 1; //total forms after addition
+        const formListContainer = this.getFormListContainer()
+
+        const newFormNode = this.createNewFormNode(newFormIndex);
+
+        formListContainer.appendChild(newFormNode)
+
+        this.setTotalFormsCount(newNumForms)
+
+        // these lines below are the only autocomplete related code
+
+        // htmx by default processes nodes it adds
+        // but since JS is adding these forms, we need to manually process them
+        htmx.process(newFormNode)
+        this.manuallyExecuteScripts(newFormNode)
+    }
+
+    manuallyExecuteScripts(container){
+        /* 
+            when you insert new scripts via innerHTML, they're not executed automatically
+            so we have to recreate these script tags and insert them via the DOM api
+        */ 
+        // Find all <script> tags in the newly inserted content
+        const scripts = container.querySelectorAll('script');
+
+        // Reinsert each <script> tag
+        scripts.forEach(originalScript => {
+            parent = originalScript.parentNode
+            const newScript = document.createElement('script');
+            // Copy the script content or src attribute
+            if (originalScript.src) {
+                newScript.src = originalScript.src; // External script
+            } else {
+                newScript.textContent = originalScript.textContent; // Inline script
+            }
+            // Copy any attributes (e.g., type, async, etc.)
+            Array.from(originalScript.attributes).forEach(attr =>
+                newScript.setAttribute(attr.name, attr.value)
+            );
+
+            // Append the new script to the document
+            parent.appendChild(newScript);
+
+        });
+
+    }
```

Comment:
The `DynamicFormsetManager` is used in a few projects now (maybe it should be formally re-used). These lines were added to support the autocomplete. It consists of,
1. activating HTMX for the nodes we're dynamically adding with JS
2. manually running scripts that are added via innerHTML

In practice this should probably be a subclass/wrapper of dynamicformsetmanager, e.g. `DynamicFormsetManagerWithAutocompleteSupport`